File: app/src/main/java/net/veldor/flibustaloader/database/dao/ReadedBooksDao.java
package net.veldor.flibustaloader.database.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import net.veldor.flibustaloader.database.entity.ReadedBooks;

import java.util.List;

@Dao
public interface ReadedBooksDao {
    @Query("SELECT * FROM ReadedBooks WHERE bookId = :id")
    ReadedBooks getBookById(String id);


    @Query("SELECT * FROM ReadedBooks")
    List<ReadedBooks> getAllBooks();

    @Insert
    void insert(ReadedBooks book);

/*    @Update
    void update(ReadedBooks book);

    @Delete
    void delete(ReadedBooks book);*/
}

File: app/src/main/java/net/veldor/flibustaloader/database/dao/ReadedBooksDao.kt
package net.veldor.flibustaloader.database.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import net.veldor.flibustaloader.database.entity.ReadedBooks

@Dao
interface ReadedBooksDao {
    @Query("SELECT * FROM ReadedBooks WHERE bookId = :id")
    fun getBookById(id: String?): ReadedBooks?

    @get:Query("SELECT * FROM ReadedBooks")
    val allBooks: List<ReadedBooks?>?

    @Insert
    fun insert(book: ReadedBooks?)
    @Delete
    fun delete(book: ReadedBooks?)
}